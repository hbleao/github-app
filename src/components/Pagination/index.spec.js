import React from 'react';
import { render } from '@testing-library/react';
import user from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import { Pagination } from '.';

describe('Pagination', () => {
  it('should be able to render the component correctly', () => {
    render(<Pagination />);

    const defaultPagination =
      document.querySelector('.pagination__link').textContent;

    expect(defaultPagination).toBe('1');
  });

  it('should be able to render the component with total pages correctly', () => {
    render(<Pagination total={10} activePage={2} />);

    const defaultPagination =
      document.querySelectorAll('.pagination__link')[4].textContent;

    expect(defaultPagination).toBe('10');
  });

  it('should be able to render the component with pagination correctly', () => {
    render(<Pagination total={10} activePage={2} />);

    const activePage = document.querySelector('.active a').textContent;

    expect(activePage).toBe('2');
  });

  it('should be able to click in the activePage', () => {
    const onClick = jest.fn();
    render(<Pagination total={10} activePage={2} onClick={onClick} />);

    const activePage = document.querySelector('.pagination__link');
    user.click(activePage);

    expect(onClick).toBeCalled();
  });

  it('should be able to render snapshot', () => {
    const component = renderer.create(
      <Pagination total={10} activePage={2} onClick={() => jest.fn()} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
