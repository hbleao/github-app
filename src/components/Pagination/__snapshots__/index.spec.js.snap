// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should be able to render snapshot 1`] = `
<ul
  className="pagination"
>
  <li
    className="pagination__item "
  >
    <a
      className="pagination__link"
      href=""
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="pagination__item active"
  >
    <a
      className="pagination__link"
      href=""
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="pagination__item "
  >
    <a
      className="pagination__link"
      href=""
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="pagination__item "
  >
    <span
      className="pagination__link"
    >
      ...
    </span>
  </li>
  <li
    className="pagination__item "
  >
    <a
      className="pagination__link"
      href=""
      onClick={[Function]}
    >
      10
    </a>
  </li>
</ul>
`;
